src/config.py

```diff
@@ -21,11 +21,24 @@ def __init__(self, s3_connection_info: Optional[S3ConnectionInfo], ingress_url:
         self.ingress_url = ingress_url
 
     _base_conf: dict[str, str] = {
-        "spark.hadoop.fs.s3a.connection.ssl.enabled": "false",
         "spark.hadoop.fs.s3a.path.style.access": "true",
         "spark.eventLog.enabled": "true",
     }
 
+    def _ssl_enabled(self) -> str:
+        """Check if ssl is enabled."""
+        if self.s3_connection_info:
+            if not self.s3_connection_info.endpoint:
+                return "true"
+            else:
+                if (
+                    self.s3_connection_info.endpoint.startswith("https:")
+                    or ":443" in self.s3_connection_info.endpoint
+                ):
+                    return "true"
+
+        return "false"
```

Comment:
I believe it is because this is the value written in the configuration file of Spark. I'm fine with either, havng the boolean parsing here or at line 64 below

Reply:
The answer done by Enrico is correct, it is a string because it is written in the spark configuration.